src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js

```diff
@@ -331,8 +331,35 @@
 
         // This is a helper method to reduce the amount of code repitition for actions: Save, Publish, SendToPublish
         function performSave(args) {
-
-
+            
+            // Check that all variants for publishing have a name.
+            if (args.action === "publish" || args.action === "sendToPublish") {
+                
+                if ($scope.content.variants) {
+                    var iVariant;
+                    for (var i = 0; i < $scope.content.variants.length; i++) {
+                        iVariant = $scope.content.variants[i];
+                        
+                        iVariant.notifications = [];// maybe not needed, need to investigate.
+                        
+                        if(iVariant.publish === true) {
+                            if (iVariant.name == null) {
+                                
+                                localizationService.localize("publish_contentPublishedFailedByMissingName").then(function (value) {
+                                    iVariant.warnings = [{"message": value.replace("%0%", iVariant.language.name)}]
```

Comment:
@nielslyngsoe I think you here can send `iVariant.language.name` as token parameter to `localizationService.localize`, then you don't need to do the replace of `%0%` - this already handled by `localize` function, so the value includes the `iVariant.language.name`.

Reply:
Perfect, just what I needed. Was thinking about looking into it but I guess I got distracted.

Reply:
Documented here: 
https://github.com/umbraco/Umbraco-CMS/blob/91c52cffc8b7c70dc956f6c6610460be2d1adc51/src/Umbraco.Web.UI.Client/src/common/services/localization.service.js#L131-L152

https://github.com/umbraco/Umbraco-CMS/blob/91c52cffc8b7c70dc956f6c6610460be2d1adc51/src/Umbraco.Web.UI.Client/src/common/directives/components/localization/localize.directive.js#L30-L47